utils/components-utils/src/main/scala/pl/touk/nussknacker/engine/util/sinkvalue/SinkValueData.scala

```diff
@@ -30,14 +36,42 @@ object SinkValueData {
   type FieldName = String
 
   sealed trait SinkValueParameter {
-    def toParameters: List[Parameter] = this match {
-      case SinkSingleValueParameter(value) => value :: Nil
-      case SinkRecordParameter(fields) => fields.values.toList.flatMap(_.toParameters)
+    def toParameters: List[Parameter] = flatten.map(_.value)
+
+    def flatten: List[SinkSingleValueParameter] = this match {
+      case single: SinkSingleValueParameter => single :: Nil
+      case SinkRecordParameter(fields) => fields.values.toList.flatMap(_.flatten)
     }
+    def validateParams(resultType: List[(String, BaseDefinedParameter)])(implicit nodeId: NodeId): ValidatedNel[ProcessCompilationError, Unit]
```

Comment:
`List` -> `Map` ?

Reply:
I also added type aliases to differentiate RecordFieldName and ParameterName